platforms/nuttx/CMakeLists.txt

```diff
@@ -138,6 +138,10 @@ if(CONFIG_NET)
 	target_link_libraries(nuttx_fs INTERFACE nuttx_net)
 endif()
 
+if(CONFIG_OPENAMP)
+	list(APPEND nuttx_libs nuttx_openamp)
+endif()
```

Comment:
move the "needed target_link_libraries" from below in here. It is the same for FLAT, PROTECTED and KERNEL builds I suppose

Reply:
moved